lua/zettelkasten/browser.lua

```diff
@@ -1,42 +1,65 @@
 local M = {}
 local fn = vim.fn
 
+local log_levels = vim.log.levels
+local log = require("zettelkasten.log")
 local config = require("zettelkasten.config")
 
-local ZK_FULL_TITLE_PATTERN = "# %d+-%d+-%d+-%d+-%d+-%d+ .+"
-local ZK_ID_PATTERN = "%d+-%d+-%d+-%d+-%d+-%d+"
-local ZK_FILE_NAME_PATTERN = "%d+-%d+-%d+-%d+-%d+-%d+.md"
-
 local s_note_cache_with_file_path = {}
 local s_note_cache_with_id = {}
 
-local function get_files(folder)
+local function get_files(folder, cfg)
```

Comment:
[non-blocking] You could just pass the `filename_pattern` here instead of passing the entire `config` object. It exposes more than this function needs and makes it just slightly harder to maintain since it's possible that this function could internally change the `config` object. Or you could just fetch the `config` internally here without relying on a parameter.

For places where multiple items from the config is accessed, and you can pass a new table with just the appropriate ones. If there isn't a lot, then just adding 1-2 more arguments is fine.